src/Runtime/OMTensorList.inc

```diff
@@ -120,3 +127,209 @@ OMTensor *omTensorListGetOmtByIndex(OMTensorList *rlist, int64_t index) {
   assert(index < rlist->_size);
   return rlist->_omts[index];
 }
+
+/* Helper function to create a sub-string between the start and end strings */
+static inline char *createSubStr(
+    char *str, const char *start, const char *end) {
+  char *startPos = strstr(str, start) + strlen(start);
+  char *endPos = strstr(str, end);
+  if (startPos != NULL && endPos != NULL && (endPos > startPos)) {
+    unsigned len = (unsigned)(endPos - startPos);
+    char *subStr = (char *)malloc(len + 1);
+    strncpy(subStr, startPos, len);
+    subStr[len] = '\0';
+    return subStr;
```

Comment:
Line 137-141 can be simplified as `return strndup(startPos, endPos - startPos);`

I think the correct code should be,
```
{
  char *startPos = strstr(str, start);
  char *endPos = strstr(str,end);
  char *p;
  return (startPos && endPos && (p = startPos + strlen(start)) < endPos) ?
    strndup(p, endPos - p) : NULL;
}

Reply:
Thanks so much! I updated the code with your suggestion.